src/frontend/src/instance/distributed.rs

```diff
@@ -509,17 +498,31 @@ impl DistInstance {
                 table_name: format_full_table_name(catalog_name, schema_name, table_name),
             })?;
 
-        let request = common_grpc_expr::alter_expr_to_request(
-            table.table_info().ident.table_id,
-            expr.clone(),
-        )
-        .context(AlterExprToRequestSnafu)?;
+        let table_id = table.table_info().ident.table_id;
+        expr.table_id = Some(api::v1::TableId { id: table_id });
 
-        let mut context = AlterContext::with_capacity(1);
+        self.verify_alter(table_id, table.table_info(), expr.clone())?;
 
-        let _ = context.insert(expr);
+        let req = SubmitDdlTaskRequest {
+            task: DdlTask::new_alter_table(expr.clone()),
+        };
 
-        table.alter(context, &request).await.context(TableSnafu)?;
+        timeout(
+            // TODO(weny): makes timeout configurable.
+            Duration::from_secs(10),
+            self.meta_client.submit_ddl_task(req),
+        )
+        .await
+        .context(error::TimeoutSnafu)?
+        .context(error::RequestMetaSnafu)?;
+
+        // Since the table information dropped on meta does not go through KvBackend, so we
+        // manually invalidate the cache here.
+        //
+        // TODO(fys): when the meta invalidation cache mechanism is established, remove it.
+        self.catalog_manager()
+            .invalidate_table(catalog_name, schema_name, table_name)
```

Comment:
Do we still need to invalidate cache here since the alter table procedure has a cache invalidation step?